twingate/client.go

```diff
@@ -105,23 +102,23 @@ type Client struct {
 }
 
 type transport struct {
-	underlyingTransport http.RoundTripper
-	APIToken            string
-	Version             string
+	underlineRoundTripper http.RoundTripper
+	APIToken              string
+	Version               string
 }
 
 func (t *transport) RoundTrip(req *http.Request) (*http.Response, error) {
-	req.Header.Add("X-API-KEY", t.APIToken)
-	req.Header.Add("User-Agent", fmt.Sprintf("TwingateTF/%s", t.Version))
+	req.Header.Set("X-API-KEY", t.APIToken)
```

Comment:
this is called again on retry and add will append an additional value on every retry